tests/Unit/PointwiseFunctions/AnalyticData/WaveEquation/Test_RegularSphericalWaveKerrSchild.cpp

```diff
@@ -0,0 +1,247 @@
+  Options<tmpl::list<RegularSphWaveKerr>> opts("");
+  opts.parse(
+      "RegularSphWaveKerr:\n"
+      "  BlackHoleMass: 1.\n"
+      "  BlackHoleSpin: [0.1,0.2,0.3]\n"
+      "  BlackHoleCenter: [0.,0.,0.]\n"
+      "  WaveProfile:\n"
+      "    Gaussian:\n"
+      "      Amplitude: 1.\n"
+      "      Width: 1.\n"
+      "      Center: 0.");
+  CHECK(opts.get<RegularSphWaveKerr>() ==
+        CurvedScalarWave::AnalyticData::RegularSphericalWaveKerrSchild(
+            1.0, {{0.1, 0.2, 0.3}}, {{0.0, 0.0, 0.0}},
+            std::make_unique<MathFunctions::Gaussian>(1., 1., 0.)));
+}
+
+void test_serialize() noexcept {
+  CurvedScalarWave::AnalyticData::RegularSphericalWaveKerrSchild solution(
+      1.0, {{0.1, 0.2, 0.3}}, {{0.0, 0.0, 0.0}},
+      std::make_unique<MathFunctions::Gaussian>(1., 1., 0.));
+  Parallel::register_derived_classes_with_charm<MathFunction<1>>();
+  test_serialization(solution);
+}
+
+void test_move() noexcept {
+  CurvedScalarWave::AnalyticData::RegularSphericalWaveKerrSchild solution(
+      1.0, {{0.1, 0.2, 0.3}}, {{0.0, 0.0, 0.0}},
+      std::make_unique<MathFunctions::Gaussian>(1., 1., 0.));
+  // since it can't actually be copied
+  CurvedScalarWave::AnalyticData::RegularSphericalWaveKerrSchild
+      copy_of_solution(1.0, {{0.1, 0.2, 0.3}}, {{0.0, 0.0, 0.0}},
+                       std::make_unique<MathFunctions::Gaussian>(1., 1., 0.));
+  test_move_semantics(std::move(solution), copy_of_solution);
+}
+
+void test_no_hole() noexcept {
+  const ScalarWave::Solutions::RegularSphericalWave flat_solution{
+      std::make_unique<MathFunctions::Gaussian>(1., 1., 0.)};
+  const CurvedScalarWave::AnalyticData::RegularSphericalWaveKerrSchild solution(
+      1., {{0., 0., 0.}}, {{1.e20, 1., 1.}},
+      std::make_unique<MathFunctions::Gaussian>(1., 1., 0.));
+
+  const tnsr::I<DataVector, 3> x{std::array<DataVector, 3>{
+      {DataVector({0., 1., 2., 3.}), DataVector({0., 0., 0., 0.}),
+       DataVector({0., 0., 0., 0.})}}};
+  auto vars = solution.variables(
+      x, 0.,
+      tmpl::list<CurvedScalarWave::Pi, CurvedScalarWave::Phi<3>,
+                 CurvedScalarWave::Psi>{});
+  auto flat_vars = flat_solution.variables(
+      x, 0., tmpl::list<ScalarWave::Pi, ScalarWave::Phi<3>, ScalarWave::Psi>{});
+  CHECK_ITERABLE_APPROX(get<CurvedScalarWave::Psi>(vars).get(),
+                        get<ScalarWave::Psi>(flat_vars).get());
+  CHECK_ITERABLE_APPROX(get<CurvedScalarWave::Pi>(vars).get(),
+                        get<ScalarWave::Pi>(flat_vars).get());
+  CHECK_ITERABLE_APPROX(get<CurvedScalarWave::Phi<3>>(vars).get(0),
+                        get<ScalarWave::Phi<3>>(flat_vars).get(0));
+  CHECK_ITERABLE_APPROX(get<CurvedScalarWave::Phi<3>>(vars).get(1),
+                        get<ScalarWave::Phi<3>>(flat_vars).get(1));
+  CHECK_ITERABLE_APPROX(get<CurvedScalarWave::Phi<3>>(vars).get(2),
+                        get<ScalarWave::Phi<3>>(flat_vars).get(2));
+}
+
+void test_kerr() noexcept {
```

Comment:
Can you add a python test for this?

Reply:
For a python test of this `AnalyticData` I would need to write up both the kerr-schild solution & flat-space regular spherical wave in python, since those solutions don't already have python implementations that they are being tested against. Is it not better if I add it as a separate PR(s) and not delay this one, because I can use their python implementations to add tests for kerr-schild and flat-space spherical wave solutions themselves in the same go then?

Reply:
Sounds good to me